pkg/controllers/user/pipeline/controller/pipelineexecution/logsyncer.go

```diff
@@ -0,0 +1,74 @@
+package pipelineexecution
+
+import (
+	"context"
+	"fmt"
+	"github.com/rancher/rancher/pkg/controllers/user/pipeline/engine"
+	"github.com/rancher/rancher/pkg/controllers/user/pipeline/utils"
+	"github.com/rancher/rancher/pkg/ticker"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	syncLogInterval = 20 * time.Second
+)
+
+type ExecutionLogSyncer struct {
+	pipelineExecutionLister    v3.PipelineExecutionLister
+	pipelineExecutionLogLister v3.PipelineExecutionLogLister
+	pipelineExecutionLogs      v3.PipelineExecutionLogInterface
+	pipelineEngine             engine.PipelineEngine
+}
+
+func (s *ExecutionLogSyncer) sync(ctx context.Context, syncInterval time.Duration) {
+	for range ticker.Context(ctx, syncInterval) {
+		logrus.Debugf("Start sync pipeline execution log")
+		s.syncLogs()
+		logrus.Debugf("Sync pipeline execution log complete")
+	}
+
+}
+
+func (s *ExecutionLogSyncer) syncLogs() {
+	Logs, err := s.pipelineExecutionLogLister.List("", utils.PipelineInprogressLabel.AsSelector())
+	if err != nil {
+		logrus.Errorf("Error listing PipelineExecutionLogs - %v", err)
+	}
+	if len(Logs) < 1 {
+		return
+	}
+	if err := s.pipelineEngine.PreCheck(); err != nil {
+		logrus.Errorf("Error get Jenkins engine - %v", err)
```

Comment:
do we want to proceed after error happens?

Reply:
I think we should mark execution as fail here instead of logging errors only.